### DISCUSSION TITLE
Choice field which allow multiple selection, is not showing inside the "Filter field" inside the "PnP - Search Filters"

### DISCUSSION THREAD

**Discussion by johnGu123** [microsoft-search/pnp-modern-search#2002] [Q&A]
We have a SharePoint list column named "category", as follow:-

![image](https://user-images.githubusercontent.com/43375450/161857542-239871e0-f339-4131-bed0-d9c730d0f0eb.png)

and it has a managed property & crawl property, as follow:-

![image](https://user-images.githubusercontent.com/43375450/161857593-379c878f-9de7-4b67-ae4f-5db42102ce09.png)

but inside the "PnP - Search Filters" web part this managed property is not showing inside the "Filter Field" any advice?
![image](https://user-images.githubusercontent.com/43375450/161857634-12a90f48-768a-4d93-90c0-a472996d6722.png)

**Comment by wobba** [comment]
Automatic managed properties are not refinable and thus cannot be used as filters. You need to map a crawled property to e.g. a RefinableString managed property to use in a filter.

**Reply by johnGu123** [reply]
@wobba  or i can make the automatic managed property as refinable ,, am i correct?

**Reply by johnGu123** [reply]
@wobba I also edit a RifinableString01 and i added the related crawl property, i re-indexed the whole site + the whole library waited now for more than 24 hours, but still the Refinable02 is not showing inside the filter web part.

here is the Refinable which i used:-

![image](https://user-images.githubusercontent.com/43375450/162092095-c7f8ecec-0ec0-4b7a-a90e-335d59bd095a.png)

any advice?

**Reply by wobba** [reply]
Check the crawl log to see if the items are re-indexed. And did you map at the site or tenant level? If at the site level, are the search parts on that site? You can also use the search query tool and see which crawled property names are part of an item.

![image](https://user-images.githubusercontent.com/51104/162254442-551171c1-b61d-4e62-b985-18a76ac34389.png)

**Reply by johnGu123** [reply]
@wobba  i did the map on the tenant level

**Reply by johnGu123** [reply]
@wobba  I am not sure if i am using the Search query tool correctly, where i defined the following : -

![image](https://user-images.githubusercontent.com/43375450/162284627-d87ff7b9-cd33-4dd0-aaf3-19b4887a07c9.png)


but i got nothing.. am i using the tool correctly ?
Thanks

**Reply by wobba** [reply]
Search for something you know the item has... title, filename, etc. Then look at the managed properties and crawled properties for the item. What you show is that no items are populated the way you expect them to be.

**Reply by johnGu123** [reply]
@wobba now i define the search query to be based on the Name to get a page named home.aspx, as follow:-

![image](https://user-images.githubusercontent.com/43375450/162543497-6a698329-e6b4-4525-a53c-bcfbe698407b.png)

where i got 5 results, but all of them are not within the site url (https://****.sharepoint.com/sites/employeecomm/....) which is shown inside the text box beside the Run command.. any advice?

**Reply by wobba** [reply]
You can use `Filename`,  with a `path` query, or just `path:https://****.sharepoint.com/sites/employeecomm/SitePages/Foo.aspx`

**Reply by johnGu123** [reply]
@wobba  ok I get the "SharePoint Search Query tool" working, where i define the path as the Query as follow:-

![image](https://user-images.githubusercontent.com/43375450/162792446-d10b8867-5a6a-4fca-a4f7-93914790ec5f.png)


and i got one result representing the page.. where i opened the "managed properties" under the "Primary results" tab and i found that the intended values for the Category column are been stored inside a managed property named "DiscussionCateogry" as follow:-

![image](https://user-images.githubusercontent.com/43375450/162792707-76b6ad6a-2dba-4aeb-ae4c-c9cb28b4fa19.png)


but i did not get the RefinableString01 inside the "managed properties" under the "Primary results" tab. so when i checked the "DiscussionCateogry" managed property inside the SharePoint online admin center , it contain the related crawl property but it is not Refinable, as follow:-

![image](https://user-images.githubusercontent.com/43375450/162793061-1c8ee493-16fc-4d6b-8e51-520cb53bcd05.png)


so i went back to "PnP Search result" web part, where i added the DisucssionCateogy to the list of managed property:-

![image](https://user-images.githubusercontent.com/43375450/162793234-4eb28cdd-5650-4511-a438-2d3b55f0ea7b.png)


and i edit the Filter web part and i added the "DiscussionCateogry" to the "Customized filters" list:-

![image](https://user-images.githubusercontent.com/43375450/162793449-f5da9b28-7154-4b7d-b748-d41a9dc0cf32.png)



but this did not show anything inside the "PnP filter web part":-

![image](https://user-images.githubusercontent.com/43375450/162793519-a54d4bb6-a97d-4538-9feb-280f4c352930.png)



could be that the "DiscussionCategory" managed property is not Refinable.. if this is the case then i am still not sure why the refinbleString01 is not showing inside the "SharePoint Search Query tool" nor inside the managed property list inside the " Pnp Search result" web part.. are you able to advice?Thanks

**Reply by wobba** [reply]
Map the same cp's to RefinableString00 and re-index your content, or edit existing content to have it re-indexed. Then you should see data.
